code/modules/spells/targeted/disorient.dm

```diff
@@ -19,4 +22,29 @@
 	compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
 	spell_flags = WAIT_FOR_CLICK
 
-	hud_state = "wiz_disorient"
\ No newline at end of file
+	hud_state = "wiz_disorient"
+	var/empowered = 0
+
+/spell/targeted/disorient/cast(var/list/targets)
+	..()
+	if(empowered)
+		for(var/mob/target in targets)
+			var/angle = pick(90, 180, 270)
+			var/client/C = target.client
+			if(C)
+				C.dir = turn(C.dir, angle)
+				spawn(30 SECONDS) //This will confuse someone for a while and end up very annoying
+				if(C)
+					C.dir = turn(C.dir, -angle)
+
+/spell/targeted/disorient/empower_spell()
+	spell_levels[Sp_POWER]++
+	empowered++
+	name = "Empowered Disorient"
+	desc = "This spell will very thoroughly disorient a target for 30 seconds."
+	return "You have upgraded the spell to turn the target's perception in another direction, further debilitating them."
+
+/spell/targeted/buttbots_revenge/get_upgrade_info(upgrade_type, level) //I don't know what this does, and it doesn't seem to do anything, just copypasted
```

Comment:
Wrong spell path

Reply:
Oh, right. I copypasted it

Reply:
that's why it doesn't do anything I guess, it handles the upgrade's description, and the description should be available via the spellbook